src/entities/addLiquidityBoosted/index.ts

```diff
@@ -0,0 +1,194 @@
+// A user can add liquidity to a boosted pool in various forms. The following ways are
+// available:
+// 1. Unbalanced - addLiquidityUnbalancedToERC4626Pool
+// 2. Proportional - addLiquidityProportionalToERC4626Pool
+import { TokenAmount } from '@/entities/tokenAmount';
+
+import { Permit2 } from '@/entities/permit2Helper';
+
+import { getAmountsCall } from '../addLiquidity/helpers';
+
+import { PoolStateWithUnderlyings } from '@/entities/types';
+
+import {
+    AddLiquidityBaseBuildCallInput,
+    AddLiquidityBaseQueryOutput,
+    AddLiquidityBuildCallOutput,
+    AddLiquidityBoostedWithOptionalInput,
+    AddLiquidityUnbalancedInputWithUserArgs,
+    AddLiquidityProportionalInputWithUserArgs,
+    AddLiquidityKind,
+} from '../addLiquidity/types';
+
+import { doAddLiquidityUnbalancedQuery } from './doAddLiquidityUnbalancedQuery';
+import { doAddLiquidityProportionalQuery } from './doAddLiquidityPropotionalQuery';
+import { Token } from '../token';
+import { BALANCER_COMPOSITE_LIQUIDITY_ROUTER } from '@/utils';
+
+import { getValue } from '@/entities/utils/getValue';
+import { encodeFunctionData } from 'viem';
+import { balancerCompositeLiquidityRouterAbi, balancerRouterAbi } from '@/abi';
+
+import { Hex } from '@/types';
+
+export class AddLiquidityBoostedV3 {
+    async query(
+        input: AddLiquidityBoostedWithOptionalInput,
+        poolState: PoolStateWithUnderlyings,
+    ): Promise<AddLiquidityBaseQueryOutput> {
+        // Technically possible to pass singleToken adds here due to type
+        // disallow it for this class
+        // TODO: Use input validator
+        const bptToken = new Token(input.chainId, poolState.address, 18);
+
+        let bptOut: TokenAmount;
+        let amountsIn: TokenAmount[];
+        let tokenInIndex: number | undefined;
+
+        // Check if userAddress and userData were provided, and assign default values if not
+        if (!('userAddress' in input) || input.userAddress === undefined) {
+            // TODO: I think using a random address might be better than using the 0 address.
```

Comment:
@johngrantuk I think defaulting to the 0 address might not be the right choice but rather a random address. What do you think?

Reply:
I think zeroAddress is best but maybe you can explain why random is better?

Reply:
My reasoning was that it is more likely for a query to return different data than what is expected as it is with a random address. This is probably an exotic scenario, so overall unlikely. So for readability / potential debugging purposes address 0 makes it easiert to work with